docs/root/faq/rolling_upgrades.rst

```diff
@@ -0,0 +1,119 @@
+.. _common_configuration_rolling_upgrades:
+
+Handling Rolling Upgrades and/or transient network failtures
+============================================================
+
+One of the biggest advantages of using Envoy in service mesh is that it frees up services 
+from implementing complex resiliency features like circuit breaking, outlier detection and retries 
+that enable services to be resilient to realities such as rolling upgrades, dynamic infrastructure, 
+and network failures. Having these features implemented at Envoy not only improves the availability 
+and resiliency of services but also brings in consistency in terms of the behaviour and observability. 
+
+This section explains at a high level the configuration supported by Envoy and how these features can be 
+used to together to handle these scenarios.
+
+**Circuit Breaking** 
+
+:ref:`Circuit Breaking<arch_overview_circuit_break>` is a critical component of distributed systems. 
+Circuit breaking lets applications configure failure thresholds that ensure safe maximums, allows components
+to fail quickly and apply back pressure as soon as possible. Applying right circuit breaking thresholds helps
+to save resources which otherwise are wasted in waiting for requests (timeouts) or retrying requests unnecessarily. 
+One of the main advantages of circuit breaking implementation in Envoy is that the circuit breaking limits are applied
+at the network level.
+
+**Retries**
+
+Automatic :ref:`request retries<config_http_filters_router>` is another measure of ensuring service resilience. Request retries should typically be used to guard against 
+transient failures. Envoy supports very rich set of configurable parameters that dictate what type of requests are retried, how many times 
+the request should be retried, timeouts for retries etc.
+
+*Retries in gRPC services*
+
+For gRPC service, envoy looks at the grpc status in the response and attempts a retry based on the statuses configured in *x-retry-grpc-on*.
+
+The following application status codes in gRPC are considered safe for automatic retry.
+
+* Cancelled - Return this code if there is an error that can be retried in the service.
+* Resource_Exhausted - Return this code if some of the resources that depends are exhausted in that instance so that retrying 
+  to another instance would help. Please note that for shared resource exhaustion, returning this will not help. Instead :ref:`rate limiting<arch_overview_rate_limit>`
+  should be used to handle such cases.
+
+The  Http Status codes 502(Bad Gateway), 503(Service Unavailable) and 504(Gateway Timeout) are all mapped to gRPC status code *Unavailable*. 
```

Comment:
nit: extra space after The
also nit: need spaces before (
nit: \*Unavailable\* -> \*UNAVAILABLE\*